config.go

```diff
@@ -329,21 +333,21 @@ func isSystem(filename string) bool {
 
 // Decode reads r into a Config, or returns an error if r could not be parsed as
 // an SSH config file.
-func Decode(r io.Reader) (*Config, error) {
+func Decode(r io.Reader, ignoreMatchDirective bool) (*Config, error) {
 	b, err := io.ReadAll(r)
 	if err != nil {
 		return nil, err
 	}
-	return decodeBytes(b, false, 0)
+	return decodeBytes(b, false, ignoreMatchDirective, 0)
```

Comment:
not really fan of passing so much params but I'm not sure why these function weren't made with an accessor. 
I can refactor if interested.